x/operator/keeper/slash.go

```diff
@@ -33,215 +33,101 @@ func GetSlashIDForDogfood(infraction stakingtypes.Infraction, infractionHeight i
 	return string(assetstype.GetJoinedStoreKey(hexutil.EncodeUint64(uint64(infraction)), hexutil.EncodeUint64(uint64(infractionHeight))))
 }
 
-func SlashFromUndelegation(undelegation *delegationtype.UndelegationRecord, totalSlashAmount, slashAmount sdkmath.Int) (sdkmath.Int, error) {
+// SlashFromUndelegation executes the slash from an undelegation
+func SlashFromUndelegation(undelegation *delegationtype.UndelegationRecord, slashProportion sdkmath.LegacyDec) (*types.SlashFromUndelegation, error) {
 	if undelegation.ActualCompletedAmount.IsZero() {
-		return totalSlashAmount, nil
+		return nil, nil
```

Comment:
It shouldn't return an error here because the `ActualCompletedAmount` will be zero when the operator suffers multiple slash events. In this case, there aren't any assets that can be slashed. This isn't an error case.